tests/end_to_end_test/api_server_repository_test.go

```diff
@@ -38,34 +38,24 @@ const (
 	controlPassword = "control-password"
 )
 
-func TestAPIServerRepository_GRPC_htpasswd(t *testing.T) {
+func TestAPIServerRepository_htpasswd(t *testing.T) {
 	t.Parallel()
 
-	testAPIServerRepository(t, []string{"--no-legacy-api"}, true, false)
+	testAPIServerRepository(t, false)
 }
 
-func TestAPIServerRepository_GRPC_RepositoryUsers(t *testing.T) {
+func TestAPIServerRepository_RepositoryUsers(t *testing.T) {
 	t.Parallel()
 
-	testAPIServerRepository(t, []string{"--no-legacy-api"}, true, true)
-}
-
-func TestAPIServerRepository_DisableGRPC_htpasswd(t *testing.T) {
-	t.Parallel()
-
-	testAPIServerRepository(t, []string{"--no-grpc", "--legacy-api"}, false, false)
+	testAPIServerRepository(t, true)
 }
 
 //nolint:thelper
-func testAPIServerRepository(t *testing.T, serverStartArgs []string, useGRPC, allowRepositoryUsers bool) {
+func testAPIServerRepository(t *testing.T, allowRepositoryUsers bool) {
 	ctx := testlogging.Context(t)
 
 	var connectArgs []string
```

Comment:
I think this is now unused/empty. Worth removing?
```suggestion
```

Reply:
done, also cleaned up below